contracts/DistributionCreator.sol

```diff
@@ -650,98 +443,179 @@ contract DistributionCreator is UUPSHelper, ReentrancyGuardUpgradeable {
+        address rewardToken
+    ) internal returns (uint256 distributionAmountMinusFees) {
+        uint256 _fees = (baseFeesValue * (BASE_9 - feeRebate[msg.sender])) / BASE_9;
+        distributionAmountMinusFees = distributionAmount;
+        if (_fees != 0) {
+            distributionAmountMinusFees = (distributionAmount * (BASE_9 - _fees)) / BASE_9;
+            address _feeRecipient = feeRecipient;
+            _feeRecipient = _feeRecipient == address(0) ? address(this) : _feeRecipient;
+            IERC20(rewardToken).safeTransferFrom(
+                msg.sender,
+                _feeRecipient,
+                distributionAmount - distributionAmountMinusFees
+            );
+        }
+        IERC20(rewardToken).safeTransferFrom(msg.sender, distributor, distributionAmountMinusFees);
+    }
+
+    /// @notice Internal version of the `sign` function
+    function _sign(bytes calldata signature) internal {
+        bytes32 _messageHash = messageHash;
+        if (ECDSA.recover(_messageHash, signature) != msg.sender) revert InvalidSignature();
+        userSignatures[msg.sender] = _messageHash;
+        emit UserSigned(_messageHash, msg.sender);
+    }
 
     /// @notice Rounds an `epoch` timestamp to the start of the corresponding period
     function _getRoundedEpoch(uint32 epoch) internal pure returns (uint32) {
-        return (epoch / EPOCH_DURATION) * EPOCH_DURATION;
-    }
-
-    /// @notice Checks whether `distribution` was live between `roundedEpochStart` and `roundedEpochEnd`
-    function _isDistributionLiveBetweenEpochs(
-        DistributionParameters memory distribution,
-        uint32 roundedEpochStart,
-        uint32 roundedEpochEnd
-    ) internal pure returns (bool) {
-        uint256 distributionEpochStart = distribution.epochStart;
-        return (distributionEpochStart + distribution.numEpoch * EPOCH_DURATION > roundedEpochStart &&
-            distributionEpochStart < roundedEpochEnd);
-    }
-
-    /// @notice Fetches data for `token` on the Uniswap `pool`
-    function _getUniswapTokenData(
-        IERC20Metadata token,
-        address pool
-    ) internal view returns (UniswapTokenData memory data) {
-        data.add = address(token);
-        data.decimals = token.decimals();
-        data.symbol = token.symbol();
-        data.poolBalance = token.balanceOf(pool);
-    }
-
-    /// @notice Fetches extra data about the parameters in a distribution
-    function getExtensiveDistributionParameters(
-        DistributionParameters memory distribution
-    ) external view returns (ExtensiveDistributionParameters memory extensiveParams) {
-        extensiveParams.base = distribution;
-        try IUniswapV3Pool(distribution.uniV3Pool).fee() returns (uint24 fee) {
-            extensiveParams.poolFee = fee;
-        } catch {
-            extensiveParams.poolFee = 0;
+        return (epoch / HOUR) * HOUR;
+    }
+
+    /// @notice Internal version of `getCampaignsBetween`
+    function _getCampaignsBetween(
```

Comment:
Shouldn't we refine the search by allowing to search for a specific campaign (or other)? Like we had before with a specific pool address?

Reply:
we have a getters for campaign by campaign Ids